docs/references/tanstack-start/get-auth.mdx

```diff
@@ -0,0 +1,117 @@
+---
+title: '`getAuth()`'
+description: The getAuth() helper retrieves the authentication state allowing you to protect your API routes or gather relevant data.
+---
+
+The `getAuth()` helper retrieves the authentication state, allowing you to protect your routes or gather relevant data.
+
+## Parameters
+
+<Properties>
+  - `request`
+
+  The Request object.
+
+  ---
+
+  - `opts?`
+
+  An optional object that can be used to configure the behavior of the `getAuth()` function. It accepts the following properties:
+
+  - `secretKey?`: A string that represents the secret key used to sign the session token. If not provided, the secret key is retrieved from the environment variable `CLERK_SECRET_KEY`.
+</Properties>
+
+## Returns
+
+`getAuth()` returns the [`Auth`](/docs/references/nextjs/auth-object) object.
+
+## Usage
+
+### Basic usage
+
+The following example demonstrates how to use `getAuth()` to retrieve authentication information in a server function and API routes.
+
+```tsx {{ filename: 'app/routes/index.tsx' }}
```

Comment:
Same errors that I mentioned from the quickstart

Reply:
As said in the quickstart this is due to the `/sign-in` route is a just placeholder to showcase that you can redirect to any defined page, we can replace the path with a comment for this example if we want to not having the type error if the path does not exist